make/tests

```diff
@@ -153,25 +209,25 @@ test/integration/multiple_translation_units_test$(EXE) : test/integration/libmtu
 ##
 # Use the stanc compiler to generate C++ from Stan programs
 ##
-test/test-models/stanc$(EXE) : O = $(O_STANC)
-test/test-models/stanc$(EXE) : test/libstanc.a
-test/test-models/stanc$(EXE) : LDLIBS += $(LDLIBS_STANC)
-test/test-models/stanc$(EXE) : %$(EXE) : %.o
+test/test-models/stanc2$(EXE) : O = $(O_STANC)
+test/test-models/stanc2$(EXE) : test/libstanc.a
+test/test-models/stanc2$(EXE) : LDLIBS += $(LDLIBS_STANC)
+test/test-models/stanc2$(EXE) : %$(EXE) : %.o
 	@mkdir -p $(dir $@)
 	$(LINK.cpp) $< $(LDLIBS) $(OUTPUT_OPTION)
 
 TEST_MODELS = $(call findfiles,src/test/test-models,*.stan)
-$(patsubst src/%.stan,%.hpp,$(TEST_MODELS)) : test/test-models/%.hpp : src/test/test-models/%.stan test/test-models/stanc$(EXE)
+$(patsubst src/%.stan,%.hpp,$(TEST_MODELS)) : test/test-models/%.hpp : src/test/test-models/%.stan $(TEST_STANC)
 	@mkdir -p $(dir $@)
-	$(WINE) test/test-models/stanc$(EXE) $< --o=$@
+	$(WINE) $(TEST_STANC) $< --o=$@
 
 ##
 # Generate C++ from Stan standalone functions
 ##
 TEST_FUNCTIONS = $(call findfiles,src/test/test-models,*.stanfuncs)
-$(patsubst src/%.stanfuncs,%.hpp,$(TEST_FUNCTIONS)) : test/test-models/%.hpp : src/test/test-models/%.stanfuncs test/test-models/stanc$(EXE)
+$(patsubst src/%.stanfuncs,%.hpp,$(TEST_FUNCTIONS)) : test/test-models/%.hpp : src/test/test-models/%.stanfuncs test/test-models/stanc2$(EXE)
 	@mkdir -p $(dir $@)
-	$(WINE) test/test-models/stanc$(EXE) $< --o=$@
+	$(WINE) test/test-models/stanc2$(EXE) $< --o=$@
```

Comment:
test/test-models/stanc2$(EXE) will be replaced by $(TEST_STANC) in this rule once stanc3 supports standaloen functions. The PR is ready but needs to be tested with rstan (no other interface supports that, so waiting on feedback). Until then this remains.